cvmfs/loader_talk.cc

```diff
@@ -136,19 +136,58 @@ int MainReload(const std::string &socket_path, const bool stop_and_go,
   }
   LogCvmfs(kLogCvmfs, kLogStdout, "done");
 
-
   // reload mode: debug (d) or non-debug (n)
-  char command = debug ? 'd' : 'n';
-  WritePipe(socket_fd, &command, 1);
-
-  command = stop_and_go ? 'S' : 'R';
-  WritePipe(socket_fd, &command, 1);
+  char commands[2];
+  commands[0] = debug ? 'd' : 'n';
+  commands[1] = stop_and_go ? 'S' : 'R';
+
+  // Loaders before version 2.11 won't recognize 'd' or 'n'. They will
+  // send "unknown command" and close the connection. We try to send
+  // both the commands and react according to what we read from the socket.
+
+  ssize_t retval;
+  do {
+    retval = send(socket_fd, commands, 2, MSG_NOSIGNAL);
+  } while ((retval <= 0) && (errno == EINTR));
+
+  if (retval <= 0) {
+    LogCvmfs(kLogCvmfs, kLogStderr, "Sending reload command failed!");
+    return 103;
```

Comment:
do we have here some constant for the return type? PANIC or something like this?

Reply:
Introducing named constants would be nice, but should be done consistently for the whole file, I'd propose to do this in a future PR.